app/services/script-tags.ts

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-mixed-spaces-and-tabs*/
+import Service, { inject as service } from '@ember/service';
+import config from 'ember-get-config';
+import Intl from 'ember-intl/services/intl';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import { MetaTagAttrs } from 'ember-osf-web/services/meta-tags';
+import getHref from 'ember-osf-web/utils/get-href';
+
+export type Content = object | string | String | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+export interface ScriptTagsData {
+    type?: DataContent;
+    content?: DataContent;
+}
+
+export interface ScriptTagDef {
+    [s: string]: DataContent;
+}
+
+export interface JSONLDScriptTagAttrs {
+    type: Content;
+}
+
+export type ScriptTagAttrs = JSONLDScriptTagAttrs;
+
+// ember-cli-meta-tags element types
+export enum TagType {
+    BASE = 'base',
+    LINK = 'link',
+    META = 'meta',
+    SCRIPT = 'script',
+    NOSCRIPT = 'noscript',
+    TITLE = 'title',
+}
+
+export interface HeadTagDef {
+    type: string;
+    content: DataContent;
+    attrs: MetaTagAttrs | ScriptTagAttrs;
+}
+
+/**
+ *
+ * Creates one script tag with model-specific overrides or default attribute values.
+ * Should additional script tags be needed, getHeadTags() should be called for each
+ * script and added to the headTags array.
+ *
+ * @method getScriptTagAttributes
+ * @method getHeadTags
+ * @param {ScriptTagsData} scriptTagsOverrides Overrides defaults
+ * @return {HeadTagsDefs} getHeadTags() returns HeadTagDefs for ScriptTagAttrs
+ */
+export default class ScriptTags extends Service {
+    @service intl!: Intl;
+    @service router!: any;
+    @service currentUser!: CurrentUserService;
+
+    async returnStructuredData(guid: string): Promise<any> {
+        const url = `${config.OSF.url}/${guid}/metadata/?format=google-dataset-json-ld`;
+        let jsonLD: object = {}; // TODO add default FE structure here
+        let jsonFetch : object | void;
+        try {
+            jsonFetch = await this.returnJSON(url);
+            if (jsonFetch && (typeof(jsonFetch) === 'object')) {
+                jsonLD = jsonFetch;
+            }
+        } catch (e) {
+            throw new Error(this.intl.t('general.structured_data.json_ld_retrieval_error'));
+        }
+        return jsonLD;
+    }
+
+    async returnJSON(url: string) {
+        const ajax = await this.currentUser.authenticatedAJAX({
+            method: 'GET',
+            url: getHref(url),
+            headers: {
+                'Content-Type': 'application/json',
+            },
+        });
+        return ajax;
+    }
```

Comment:
The logic in this snippet is a bit hard to follow. In `returnStructuredData`, we first construct a `url`, and then pass it to `returnJSON()`. `returnJSON` returns the result of the the `authenticatedAjax` call to the `url` and assign it to `jsonFetch`. if `jsonFetch` is not falsy and is an `object`, we assign `jsonFetch` to `jsonLD` and then return.

There are several things worth mentioning:

1. `getHref` in line 78 is redundant, since it would always return the parameter if the parameter is a string.
2. Checking `typeof (jsonFetch) === 'object'` is also redundant. In JS, the `typeof` operator only has several possible return values. If the operand is a primitive type, then it would return its type (`string`, `number`, and etc.) If it is a symbol, then it will return `'symbol'`. For functions it returns `'function'` and for anything else (including `null`) it returns `'object'`. Since `authenticateAJAX` is a wrapper of `$.ajax()`, `jsonFetch` would always have type of the resolved value of the `$.ajax()` call. The type of the resolved value depends on the `Content-Type` header from the server, and in this case it is `application/ld+json`. Therefore `typeof jsonFetch` would always be `'object'`.
3. Assigning `jsonFetch` to `jsonLD` serves no real purpose since `jsonLD` and `jsonFetch` are pointers to the same underlying object.
4. We set the `Content-Type` header of a request because it is a `POST` or `PUT` request that may contain additional information in the request body. The server needs to know what parser to use to decode the information. In this case, it is a `GET` request and shouldn't have anything in the body.
5. Usually, when we `throw` an Error, we expect some upper-level error handler to handle the thrown exception. There does not seem to be any newly defined error handler. And I don't know how native Ember route error handlers would react to uncaught exceptions like this.

I believe the purpose of fetching the json-ld metadata could be done using just a couple of lines. See comment below.

Reply:
1. getHref is now removed.
2.,3. I've simplified the jsonFetch logic and included possibilities for null or undefined when there is no object returned. I've also added a check to see if not simply an object, but if entries are contained in the object. 
4. I have removed the Content-Type for the GET request.
5. I have replaced the throw new Error with the captureException(e) method.